### PR TITLE
Improve the local generation of metadata to support registrationAutho…

### PR DESCRIPTION
**Pull request by br00k**
…rity

Useful in the leadup to entity category support to publish
registrationAuthority (which might be overridden by a federation) and
attributes.required.

This was half of #49 and should be unbundled from that request.

### CODE REVIEW

File: lib/SimpleSAML/Metadata/SAMLParser.php [hunk 1]
```diff
@@ -478,6 +484,12 @@ public function getMetadata1xSP() {
 		if (array_key_exists('attributes', $spd)) {
 			$ret['attributes'] = $spd['attributes'];
 		}
+		if (array_key_exists('attributes.required', $spd)) {
+			$ret['attributes.required'] = $spd['attributes.required'];
+		}
+		if (array_key_exists('attributes.required', $spd)) {
+			$ret['attributes.required'] = $spd['attributes.required'];
+		}
```
Comment (thijskh) [inline lib/SimpleSAML/Metadata/SAMLParser.php]: This block is duplicated?

**Comment by jaimeperez** [comment]
Just for the record, so that Brook can complain about me denying his pull requests (:wink:) , support for `registrationAuthority` (and *RegistrationInfo* in general as an extension) is already in place.

**Comment by thijskh** [comment]
Given the back-and-forths it may be useful to squash the commits before merging.

**Comment by thijskh** [comment]
:+1:

**Comment by jaimeperez** [comment]
Merged! Just because it was you, Brook :wink:

**Pull request merged**